cmake/modules/SwiftSource.cmake

```diff
@@ -451,12 +452,24 @@ function(_compile_swift_files
 
   # Then we can compile both the object files and the swiftmodule files
   # in parallel in this target for the object file, and ...
+
+  # Windows doesn't support long command line paths, of 8191 chars or over.
+  # We need to work around this by avoiding long command line arguments. This can be
+  # achieved by writing the list of file paths to a file, passing that file to a new
+  # Python utility script (line-directive-file), which reads the list of file paths,
+  # passing them to the old Python utility script, line-directive.
+  set(line_directive_tool "${SWIFT_SOURCE_DIR}/utils/line-directive-file")
+  set(file_path "${CMAKE_CURRENT_BINARY_DIR}/line-directive.tmp")
```

Comment:
I'm not sure that I like this.  Just feels like the could be a collision of there are parallel builds.  Can we use a generated name?